packages/core/src/util/cell.ts

```diff
@@ -109,17 +110,22 @@ export const mapStateToCellProps = (
     ownProps.visible !== undefined
       ? ownProps.visible
       : isVisible(uischema, rootData, undefined, getAjv(state));
-  const readonly = state.jsonforms.readonly;
-  const enabled =
-    !readonly && (ownProps.enabled !== undefined
-      ? ownProps.enabled
-      : isEnabled(uischema, rootData, undefined, getAjv(state)));
   const errors = formatErrorMessage(
     union(getErrorAt(path, schema)(state).map(error => error.message))
   );
   const isValid = isEmpty(errors);
   const rootSchema = getSchema(state);
 
+  const config = getConfig(state);
+  const enabled: boolean = isInherentlyEnabled(
+    state,
+    ownProps,
+    uischema,
+    schema || rootSchema,
+    rootData,
+    config
+  );
```

Comment:
For cells we should keep the old behavior. The old algorithm preferred `ownProps` over `rules`. This makes sense as we don't distinguish between cells. Either all cells are enabled or all cells are disabled. The Table renderer by default just passes its own `enabled` prop to all cells.

Therefore the enabled calculation via rules for cells is just a fallback for non-default table renderers which for some reason don't pass on the `enabled` prop. We should prefer the passed on prop (i.e. `ownProp`) as otherwise the same ui schema rule might be evaluated again and again for each of the cells. This will get really expensive for large tables as validating is not cheap.
```suggestion
  const config = getConfig(state);
  const enabled: boolean = typeof ownProps.enabled === 'boolean' ? ownProps.enabled : isInherentlyEnabled(
    state,
    ownProps,
    uischema,
    schema || rootSchema,
    rootData,
    config
  );
```

The old cell tests actually tested for this and need to be restored.